docs/conf.py

```diff
@@ -102,14 +102,21 @@
 # documentation created by autosummary uses a template file (in autosummary in the templates path),
 # which likely overrides the autodoc defaults.
 
+# These options impact when using `.. autoclass::` manually.
+# They do not impact the `.. autosummary::` templates.
+autodoc_default_options = {
+    "show-inheritance": True,
+}
+
 # Move type hints from signatures to the parameter descriptions (except in overload cases, where
 # that's not possible).
 autodoc_typehints = "description"
-# Only add type hints from signature to description body if the parameter has documentation.  The
-# return type is always added to the description (if in the signature).
-autodoc_typehints_description_target = "documented_params"
-
```

Comment:
Is there a chance we are exposing parameters that were being kept private on purpose with this change? Else I wonder why this variable would have been set originally.

Reply:
When we swapped from using the third-party `sphinx-autodoc-typehints` to the built-in `sphinx.ext.autodoc` support for this, I think I just set the options that caused the most similar behaviour to before (#9705).

Reply:
> Is there a chance we are exposing parameters that were being kept private on purpose with this change?

It wouldn't make sense to me to have certain parameters be private and others public. I suspect certain parameters aren't being documented simply due to oversight, or perhaps people thinking it's intuitive so it doesn't warrant docstring.

Also, the parameters _are_ already in the docs as part of the signature. The only change this PR makes is showing their type hints.